cypress/e2e/homepage.spec.ts

```diff
@@ -0,0 +1,217 @@
+          "be.visible"
+        );
+      }
+
+      cy.findByTestId("statistic-title-TOTAL DFI MINTED").should("be.visible");
+      cy.findByTestId("statistic-title-MASTERNODES").should("be.visible");
+    });
+
+    // No TC
+    it("should have clickable Explore DeFi Meta Chain button", () => {
+      cy.findByTestId("explore-dmc-button-Explore DeFi Meta Chain")
+        .should("be.visible")
+        .and("have.attr", "href")
+        .and("include", "https://meta.defichain.com");
+    });
+
+    // No TC
+    it("should have clickable View all articles button", () => {
+      cy.findByTestId("explore-dmc-button-View all articles")
+        .should("be.visible")
+        .and("have.attr", "href")
+        .and("include", "https://blog.defichain.com");
+    });
+
+    // No TC
+    it("should have Ecosystem links displayed", () => {
+      ecosystemLinks.forEach((ecosystemItem) => {
+        cy.findByTestId(
+          `ecosystem-link-${ecosystemItem.testid}`
+        ).scrollIntoView();
+        cy.findByTestId(`ecosystem-link-${ecosystemItem.testid}`)
+          .should("contain", ecosystemItem.title)
+          .and("be.visible")
+          .and("have.attr", "href")
+          .and("include", ecosystemItem.link);
+      });
+    });
+
+    // No TC
+    it("should have expected Blockchain feature section title and description", () => {
+      cy.findByTestId("section-title-NON-TURING-COMPLETE BLOCKCHAIN").should(
+        "be.visible"
+      );
+      cy.findByTestId(
+        "section-header-Designed,for,Decentralized,Finance"
+      ).should("be.visible");
+      cy.findByTestId("section-desc-Designed,for,Decentralized,Finance")
+        .should("be.visible")
+        .and(
+          "have.text",
+          "DeFiChain offers comprehensive functionality tailored to the Distributed Ledger Technology (DLT) " +
+            "community, with an emphasis on simplicity, speed, and security."
+        );
+    });
+
+    // No TC
+    it("should have Blockchain features section displayed", () => {
+      const textAreas = ["title", "desc"];
+      const features = [
+        "THROUGHPUT",
+        "SECURITY",
+        "IMMUTABILITY",
+        "DEVELOPMENT",
+        "VARIETY",
+      ];
+      textAreas.forEach((area) => {
+        features.forEach((feature) => {
+          if (viewport === "iphone-xr" && feature === "DEVELOPMENT") {
+            cy.get("ul.blockchain-features-dots > li > button")
+              .last()
+              .click({ force: true }); // covered by div.mx-6 md:mx-12 xl:mx-[120px]
+          }
+          cy.findAllByTestId(`highlights-${feature}-${area}`).should(
+            "be.visible"
+          );
+        });
+      });
+    });
+
+    // No TC
+    it("should have Blog posts with links displayed", () => {
+      cy.findAllByTestId("blog-post-container-link")
+        .should("be.visible")
+        .and("have.attr", "href")
+        .and("include", "https://blog.defichain.com/");
```

Comment:
for link that is navigating out from the domain, let's add a check to make sure `target = _blank`